src/main/java/game/Scene.java

```diff
@@ -22,61 +51,104 @@
-
-            StringBuilder message = new StringBuilder("inspect:\n");
-            message.append(name).append("\n\n");
-            message.append(description).append("\n\n");
-            message.append("scene commands:\n");
-            listCommands(command.getGame()).forEach(s -> message.append("\t").append(s).append("\n"));
-            message.append("player commands:\n"); // TODO: 06-03-2020 handle listing player commands somewhere else, this is messy
-            command.getGame().getPlayer()
-                    .listCommands(command.getGame()).forEach(s -> message.append("\t").append(s).append("\n"));
-            message.append("\n\n");
+            } else {
+                callback.onMessage(inspect(command));
+            }
 
-            callback.onMessage(message.toString());
             return true;
-        } else if (command.getAction().equals("search")) {
-            StringBuilder message = new StringBuilder("search:\n");
-            message.append("You find the following items in the scene:\n");
-            items.forEach(i -> message.append("\t").append(i).append("\n"));
-            message.append("\n\n");
+        } else if (command.getAction().equals("search")) { // handles search command
 
-            callback.onMessage(message.toString());
+            callback.onMessage(search(command));
             return true;
-        } else if (actions != null && !command.hasReceiver() && actions.hasCommand(command.getAction())) {
+        } else if (actions != null && !command.hasReceiver() && actions.hasCommand(command.getAction())) { // handles scene action
             // TODO: 04-03-2020 replace with Action.onCommand()
-            Effect effect = actions.getEffect(command.getAction());
-            if (effect != null) {
+            Effects effects = actions.getEffects(command.getAction());
+            if (effects != null) {
 
-                effect.apply(command.getGame());
+                effects.apply(command.getGame(), callback);
 
-                callback.onMessage("Effect applied... " + effect.toString());
+                callback.onMessage("Effect applied... " + effects.toString());
             }
             return true;
-        } else if (command.hasReceiver()) {
-            if (items.contains(command.getReceiver())) {
-                Item item = command.getGame().getItemMap().get(command.getReceiver());
-                item.onCommand(command, callback);
+        } else if (command.hasReceiver() && command.getAction().equals("go")) { // handles next scene action
+            if (nextScenes.containsKey(command.getReceiver())) {
+                command.getGame().setCurrentSceneById(nextScenes.get(command.getReceiver()), callback);
             } else {
-                callback.onMessage("This item does not exist.");
+                callback.onMessage("This direction is not known...");
             }
 
             return true;
```

Comment:
That's one big ass if-else statement handling XD. Maybe you can reduce these into smaller functions that you call in the handle command function. You can then put all the handling in different functions and then call them one by one inside this function to make it simpler. The if-else would get smaller too and it would be clearer what this big function does by dividing its task into smaller functions. I don't know if that would be possible tho. This looks a bit difficult to read but I can understand/see why it is hard to do it in another way. Good job anyway tho and Great job in handling all possible scenarios!